src/main/java/org/sobotics/redunda/DataService.java

```diff
@@ -0,0 +1,433 @@
+	 * */
+	private List<String> trackedFiles = new ArrayList<String>();
+	
+	/**
+	 * Initialize with default values
+	 * */
+	public DataService() {}
+	
+	/**
+	 * Initializes the object with the api key
+	 * 
+	 * @param key The API key for Redunda. You can get it form the instances overview
+	 * */
+	public DataService(String key) {
+		this.apiKey = key;
+	}
+	
+	/**
+	 * Sets the time between two syncs. (if your bot syncs automatically)
+	 * 
+	 * @param seconds The new value
+	 * */
+	public void setSyncInterval(int seconds) {
+		this.interval = seconds > 0 ? seconds : 180;
+	}
+	
+	/**
+	 * Returns the time in seconds between two syncs.
+	 * 
+	 * @return The value in seconds
+	 * */
+	public int getSyncInterval() {
+		return this.interval;
+	}
+	
+	/**
+	 * Adds a file to the list of tracked files
+	 * 
+	 * @param path The path to the file to track
+	 * */
+	public void trackFile(String path) {
+		if (this.trackedFiles.contains(path)) {
+			System.out.println("Already tracking "+path);
+		} else {
+			this.trackedFiles.add(path);
+		}
+	}
+	
+	/**
+	 * Returns the list of tracked files
+	 * 
+	 * A tracked file may NOT contain the string `_slash_`!
+	 * 
+	 * @return The list of tracked files
+	 * */
+	public List<String> getTrackedFiles() {
+		return this.trackedFiles;
+	}
+	
+	/**
+	 * Encodes a filename
+	 * 
+	 * @param filename The filename to encode
+	 * 
+	 * @return The encoded filename
+	 * */
+	private String encodeFilename(String filename) {
+		return filename.replace("/", "_slash_");
+	}
+	
+	/**
+	 * Decodes a filename
+	 * 
+	 * @param filename The filename to decode
+	 * 
+	 * @return The decoded filename
+	 * */
+	private String decodeFilename(String filename) {
+		return filename.replace("_slash_", "/");
+	}
+	
+	/**
+	 * Uploads all tracked files to Redunda.
+	 * 
+	 * This will ALWAYS overwrite the files on the server
+	 * 
+	 * @throws IOException If the upload of a file fails
+	 * */
+	public void pushFiles() throws IOException {
+		for (String file : this.trackedFiles) {
+			this.pushFile(file);
+		}
+	}
+	
+	/**
+	 * Uploads a file to Redunda
+	 * 
+	 * This will ALWAYS overwrite the files on the server
+	 * 
+	 * @param filename The name of the file to upload
+	 * @throws IOException if the file couldn't be read
+	 * */
+	public void pushFile(String filename) throws IOException {
+		String content = new String(Files.readAllBytes(Paths.get(filename)));
+		System.out.println(content);
+		String encodedFilename;
+		try {
+			encodedFilename = URLEncoder.encode(this.encodeFilename(filename), "UTF-8");
+		} catch (Throwable e) {
+			e.printStackTrace();
+			return;
+		}
+		
+		
+		String url = "https://redunda.sobotics.org/bots/data/"+encodedFilename+"?key="+this.apiKey;
+		URL obj = new URL(url);
+		HttpsURLConnection con = (HttpsURLConnection) obj.openConnection();
+
+		//add request header
+		con.setRequestMethod("POST");
+		con.setRequestProperty("User-Agent", "Redunda Library");
```

Comment:
This doesn't conform to the User-Agent spec: `requester-name/requester-version [extra-info]`. Maybe something like `redunda-lib-java/1.7`?